docker-compose-single.yml

```diff
@@ -14,4 +16,5 @@ services:
       - "1337:1337"
       - "8080:8080"
       - "8180:8180"
+    init: true
```

Comment:
Could you explain why this flag is necessary?

Reply:
There is no init system in docker and therefore, first process that is executed (`./iroha` in our case) gets PID = 1. This PID is special because it ignores signals sent to it like SIGKILL or SIGTERM. You can see this if you try to kill the process from inside container(and you cannot) or you can see that running `docker stop` takes 10sec to complete because it waits on graceful shutdown which isn't happening. 

This can be resolved in two ways, either by adding small init system called `tini` which is packed with docker or explicit signal handling needs to be included in the iroha binary(I haven't confirmed this)

Reply:
Explicit signal handling is planned after #1480. With that said, we don't gracefully exit regardless, so the implicit signal handling is fine for now.